cmd/k8s-deployment-scaler/main.go

```diff
@@ -335,13 +331,26 @@ func listDeployments(w http.ResponseWriter, r *http.Request) {
 	namespace := r.URL.Query().Get("namespace")
 
 	var deployments []string
+	var err error
+	var list []*appsv1.Deployment
 
-	// Use the informer's cache to list deployments
-	for _, obj := range deploymentInformer.GetIndexer().List() {
-		deployment := obj.(*appsv1.Deployment)
-		if namespace == "" || deployment.Namespace == namespace {
-			deployments = append(deployments, fmt.Sprintf("%s/%s", deployment.Namespace, deployment.Name))
-		}
+	if namespace == "" {
+		list, err = deploymentLister.List(labels.Everything())
+	} else {
+		list, err = deploymentLister.Deployments(namespace).List(labels.Everything())
+	}
```

Comment:
```suggestion
	list, err := deploymentLister.Deployments(namespace).List(labels.Everything())
```

The lister will returns all deployments if namespace is `""`.

Reply:
Thank you, I have implemented your suggestion.